static/js/portico/signup.js

```diff
@@ -135,18 +135,20 @@ $(function () {
     }
 
     function update_full_name_section() {
-        if ($("#source_realm_select").length && $("#source_realm_select").find(":selected").val() !== "on") {
-            $("#full_name_input_section").hide();
-            $("#profile_info_section").show();
-            var avatar_url = $("#source_realm_select").find(":selected").attr('data-avatar');
-            var full_name = $("#source_realm_select").find(":selected").attr('data-full-name');
-            $("#profile_full_name").text(full_name);
-            $("#id_full_name").val(full_name);
-            $("#profile_avatar").attr("src", avatar_url);
-        } else {
-            $("#full_name_input_section").show();
-            $("#profile_info_section").hide();
-            $("#id_full_name").val("");
+        $("#full_name_input_section").show();
+        $("#profile_info_section").hide();
+        if ($("#source_realm_select").length) {
+            if ($("#source_realm_select").find(":selected").val() !== "on") {
+                $("#full_name_input_section").hide();
+                $("#profile_info_section").show();
+                var avatar_url = $("#source_realm_select").find(":selected").attr('data-avatar');
+                var full_name = $("#source_realm_select").find(":selected").attr('data-full-name');
+                $("#profile_full_name").text(full_name);
+                $("#id_full_name").val(full_name);
+                $("#profile_avatar").attr("src", avatar_url);
+            } else {
+                $("#id_full_name").val("");
+            }
```

Comment:
Can you explain the effect of this more clearly?

Reply:
Explained in the commit message. Also as you mentioned this commit fixes #10867.